psycopg/.flake8

```diff
@@ -0,0 +1,6 @@
+[flake8]
```

Comment:
Is this file really needed? It seems redundant with the one at root level.

Reply:
I have to double-check if they are still both needed. Until recently (where this config was in the `tox.ini` files), the top-level flake8 file was taken into account when running `flake8` from the root of the project, to check all the code, but, when the vim plugin checks the syntax (using `flake8 /absolute/path/to/file.py` I assume), I understand that flake8 stops at the first `setup.py` or similar file it finds, which in most projects identify the root of the project, but not in this "monorepo".

I hate the duplication too. I'll check if I can drop it.

Reply:
If the top file is dropped, flake8 gets lost in the `.venv` directories. It doesn't have the smarts to check in the `.gitignore`.

```
$ flake8  # finishes quickly
$ rm .flake8 
$ flake8  # several seconds of inactivity
^C... stopped while processing files

$ flake8 -v
flake8.checker            MainProcess     46 INFO     Making checkers
^Cflake8.checker            MainProcess  28457 WARNING  Flake8 was interrupted by the user
... stopped while processing files
```
so that's kinda needed. Trying to drop the directory ones, vim doesn't pick up on the config:

![image](https://user-images.githubusercontent.com/199429/208316148-9aed4942-e967-4a48-9ecf-a587636061a9.png)

I use vim with [ALE](https://github.com/dense-analysis/ale). Using `:ALEInfo` it shows that the command line executed is:
``` 
['/bin/bash', '-c', 'cd ''/home/piro/dev/psycopg3/psycopg'' && ''/home/piro/dev/psycopg3/.venv/bin/flake8''
--format=default --stdin-display-name ''/home/piro/dev/psycopg3/psycopg/psycopg/errors.py'' - < ''/tmp/vOedQyL/4248/errors.py''']
```
the problem with this is the `cd` to the `psycopg` sub-directory of the project instead of the root:

```
(.venv) piro@sherekhan:~/dev/psycopg3$ flake8 /home/piro/dev/psycopg3/psycopg/psycopg/errors.py
# finds no problem

(.venv) piro@sherekhan:~/dev/psycopg3$ cd psycopg
/home/piro/dev/psycopg3/psycopg
(.venv) piro@sherekhan:~/dev/psycopg3/psycopg$ flake8 /home/piro/dev/psycopg3/psycopg/psycopg/errors.py 
# finds the problem
```

I'll check if I can configure ALE better...

Reply:
Ale, as I understood, takes the `psycopg` directory as the root as it looks [one of these files](https://github.com/dense-analysis/ale/blob/8bcbb2995822c4062d00cb41df713fd76092f04f/doc/ale-python.txt#L24).  There seem to be a way to customize it [using `ale_root`](https://github.com/dense-analysis/ale/blob/8bcbb2995822c4062d00cb41df713fd76092f04f/doc/ale.txt#L1953) but I'm not so much an expert of vim so it will take some time to figure out how to write a function e.g. to use `.git` as the project root.

Reply:
Asked in https://github.com/dense-analysis/ale/discussions/4391

Reply:
Indeed I also have trouble within vim when the inner file is dropped. That's for the investigation.
But maybe the file could be a symlink?